src/connectionconfig/connectionDialogWebViewController.ts

```diff
@@ -184,96 +180,75 @@ export class ConnectionDialogWebViewController extends ReactWebViewPanelControll
-						};
-					}
-					return {
-						isValid: true,
-						validationMessage: ''
-					};
-				}
-			},
-			{
-				type: FormComponentType.Dropdown,
-				propertyName: 'authenticationType',
-				label: 'Authentication Type',
-				required: true,
-				options: [
-					{
-						displayName: 'SQL Login',
-						value: AuthenticationType.SqlLogin
-					},
-					{
-						displayName: 'Windows Authentication',
-						value: AuthenticationType.Integrated
-					},
-					{
-						displayName: 'Microsoft Entra MFA',
-						value: AuthenticationType.AzureMFA
-					}
-				],
-			},
-			{
-				// Hidden if connection string is set or if the authentication type is not SQL Login
-				propertyName: 'user',
-				label: 'User Name',
-				type: FormComponentType.Input,
-				required: true,
-				validate: (value: string) => {
-					if (this.state.connectionProfile.authenticationType === AuthenticationType.SqlLogin && !value) {
+	private convertToFormComponent(connOption: ConnectionOption): FormComponent {
+		switch (connOption.valueType) {
+			case 'boolean':
+				return {
+					propertyName: connOption.name as keyof IConnectionDialogProfile,
+					label: connOption.displayName,
+					required: connOption.isRequired,
+					type: FormComponentType.Checkbox,
+					tooltip: connOption.description,
+				};
+			case 'string':
+				return {
+					propertyName: connOption.name as keyof IConnectionDialogProfile,
+					label: connOption.displayName,
+					required: connOption.isRequired,
+					type: FormComponentType.Input,
+					tooltip: connOption.description,
+				};
+			case 'password':
+				return {
+				propertyName: connOption.name as keyof IConnectionDialogProfile,
+				label: connOption.displayName,
+				required: connOption.isRequired,
+				type: FormComponentType.Password,
+				tooltip: connOption.description,
+			};
+
+			case 'number':
+				return {
+					propertyName: connOption.name as keyof IConnectionDialogProfile,
+					label: connOption.displayName,
+					required: connOption.isRequired,
+					type: FormComponentType.Input,
+					tooltip: connOption.description,
+				};
+			case 'category':
+				return {
+					propertyName: connOption.name as keyof IConnectionDialogProfile,
+					label: connOption.displayName,
+					required: connOption.isRequired,
+					type: FormComponentType.Dropdown,
+					tooltip: connOption.description,
+					options: connOption.categoryValues.map(v => {
 						return {
-							isValid: false,
-							validationMessage: 'User name is required'
+							displayName: v.displayName ?? v.name, // Use name if displayName is not provided
+							value: v.name
 						};
-					}
-					return {
-						isValid: true,
-						validationMessage: ''
-					};
-				}
-			},
-			{
-				propertyName: 'password',
-				label: 'Password',
-				required: false,
-				type: FormComponentType.Password,
-			},
-			{
+					}),
+				};
+			default:
+				console.log(`Unhandled connection option type: ${connOption.valueType}`);
```

Comment:
should we be using `Logger` for error messages?  I see inconsistent usage of Logger and accessing console directly through the codebase, but that doesn't seem desirable unless I'm missing something.